submissions/UA-girl/task_memory_game/js/index.js

```diff
@@ -0,0 +1,127 @@
+const images = [
+    {
+        name: 'apple-green',
+        src: 'img/apple-green.svg'
+    },
+    {
+        name: 'blackberry',
+        src: 'img/blackberry.svg'
+    },
+    {
+        name: 'blueberry',
+        src: 'img/blueberry.svg'
+    },
+    {
+        name: 'cherry',
+        src: 'img/cherry.svg'
+    },
+    {
+        name: 'grape',
+        src: 'img/grape.svg'
+    },
+    {
+        name: 'lingonberry',
+        src: 'img/lingonberry.svg'
+    }
+];
+const TIME_DELAY = 400;
+const container = document.querySelector('.container-game');
+let cards = [];
+let starTime, endTime;
+
+
+function shuffleArray(array) {
+    for (let i = array.length - 1; i > 0; i--) {
+        const j = Math.floor(Math.random() * (i));
+        [array[i], array[j]] = [array[j], array[i]];
+    }
+}
+
+
+function createCardItem({name, src}) {
+    const cardItem = document.createElement('div');
+    cardItem.classList.add('card');
+    cardItem.setAttribute('data-name', name);
+    const div_front = document.createElement('div');
+    div_front.classList.add('front');
+    const img = document.createElement('img');
+    img.classList.add('card-image');
+    img.setAttribute('src', src);
+    div_front.appendChild(img);
+    const div_back = document.createElement('div');
+    div_back.classList.add('back');
+    cardItem.appendChild(div_front);
+    cardItem.appendChild(div_back);
+    return cardItem;
+}
+
+
+function updateGame() {
+    shuffleArray(cards);
+    cards.forEach(card => container.appendChild(card));
```

Comment:
DOM operations are very expensive and should be batched if possible. Here instead of appending 12 card one by one you can create a wrapper (take a look at `document.createDocumentFragment` API), append cards to it and then attach this wrapper to a DOM node. this will result in 1 DOM access instead of 12.

Reply:
Thank you! Fixed )